apps/mobile/src/state/chat/utils/offerStates.ts

```diff
@@ -0,0 +1,61 @@
+import {type ChatWithMessages, type RequestState} from '../domain'
+import {DateTime} from 'luxon'
+import {type OfferRerequestLimitDays} from '../../../utils/remoteConfig/domain'
+
+export function getRequestState(chat?: ChatWithMessages): RequestState {
```

Comment:
In this case I decided to use playin functions instead of atoms, because sometimes we need these values when we have atoms sometimes we need them when we have values directly. This the most clean approach I was able to find. There are few atoms created from these functions below.